tools/util/OperationManager.hpp

```diff
@@ -0,0 +1,1172 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#ifndef REPLAY_OperationManager_HPP
+#define REPLAY_OperationManager_HPP
+
+#include <iostream>
+#include <cstdint>
+#include <vector>
+
+#include "umpire/Allocator.hpp"
+#include "umpire/strategy/AllocationAdvisor.hpp"
+#include "umpire/strategy/SizeLimiter.hpp"
+#include "umpire/strategy/MixedPool.hpp"
+#include "umpire/strategy/MonotonicAllocationStrategy.hpp"
+#include "umpire/strategy/SlotPool.hpp"
+#include "umpire/strategy/ThreadSafeAllocator.hpp"
+#include "umpire/ResourceManager.hpp"
+
+class Operation {
```

Comment:
Can we call this `ReplayOperation` to avoid overloading the term? We use "operations" elsewhere in Umpire.

Reply:
Done